code/modules/antagonists/_common/antag_datum.dm

```diff
@@ -164,11 +164,9 @@ GLOBAL_LIST_EMPTY(antagonists)
 /datum/antagonist/proc/remove_blacklisted_quirks()
 	var/mob/living/L = owner.current
 	if(istype(L))
-		var/list/my_quirks = L.client?.prefs.all_quirks.Copy()
-		SSquirks.filter_quirks(my_quirks,blacklisted_quirks)
 		for(var/q in L.roundstart_quirks)
 			var/datum/quirk/Q = q
-			if(!(SSquirks.quirk_name_by_path(Q.type) in my_quirks))
+			if(Q.type in blacklisted_quirks)
 				if(initial(Q.antag_removal_text))
 					to_chat(L, "<span class='boldannounce'>[initial(Q.antag_removal_text)]</span>")
 				L.remove_quirk(Q.type)
```

Comment:
While that was already a thing before this PR, probably should  just qdel the quirk considering remove_quirk goes though the roundstart quirk list *again* to qdel it and does nothing else, which is kindaa unneeded if we already have the quirks we want to del.